modules/general_functions.py

```diff
@@ -656,3 +657,51 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
         return next(i for i in iterable if cond(i))
     except StopIteration:
         raise ValueError("Value not found in iterable.")
+
+# FIXME: Find better placement
+class ReferenceDensity:
```

Comment:
`ReferenceDensity` could be in a new file (**modules/reference_density.py**).

Reply:
OK